### ISSUE TITLE
Compiler error on Mac

### ISSUE THREAD

**Issue by trungtt1981** [miniupnp/miniupnp#271]
I can compile and build successfully on Linux. 
However, on Mac, always get the following compiler error:

minissdp.c:878:7: error: redefinition of 'cmbuf' with a different type: 'char [32]' vs 'char [24]'
        char cmbuf[CMSG_SPACE(sizeof(struct sockaddr_dl))];

Build command: INCLUDES="-I.../xnu/bsd -I.../xnu/libkern" make -f Makefile.macosx

Root cause is because both of the flags "IP_RECVIF" and "IP_PKTINFO" are defined in "xnu/bsd"

**Comment by miniupnp** [comment]
Mac OS X is no more supported.
Apple is preventing 3rd party applications to interact with pf

**Comment by trungtt1981** [comment]
That's really bad new for me.
If possible, please kindly provide me some link to download the pre-built executable miniupnpd program for Mac OS 10.12.6 or some alternative one? 
So many thanks !

**Comment by miniupnp** [comment]
What are you not understanding in "Mac OS X is not supported anymore" ?

anyway nobody is using mac OS X as a home router, do you ?

**Comment by trungtt1981** [comment]
I understand from your answer that it's no longer possible to build miniupnpd on Mac. However, I still want to check if some prebuilt on Mac in the past still available so that I can download and try.
My computer at work is Mac OS while my staging server is Ubuntu. On my Mac, I do need to have something like miniupnpd for running the UPnP which is needed by the Ethereum node.

**Comment by miniupnp** [comment]
no pre-built miniupnpd is available here. You may want to check your favorite package manager (homebrew ?)

Anyway, you don't seams to understand that UPnP IGD has to be provided by your router, not your workstation... Also in an office LAN environment, it is wiser not to use UPnP and have your network administrator manage NAT port mappings

**Comment by trungtt1981** [comment]
I understand that the UPnP IGD is provided by most of the routers. 
My current work is to deploy on my own Mac computer a private net of Ethereum nodes that use P2P networking based on your library libminiupnpc. Each private Ethereum node acts as a UPnP client and thus I need to have some UPnP daemon run in background. So, I have tried to build and run your miniupnpd daemon on my own Mac computer. But it cannot be built on Mac. I have found a good alternative that is "Serviio". However, this alternative doesn't offer to set the "external IP" as for your miniupnpd.conf. 

On the staging server of Ubuntu, I can build and run quite well your miniupnpd.
